pkg/storage/tsdb/config.go

```diff
@@ -15,12 +15,13 @@ const (
 
 // Config holds the config information for TSDB storage
 type Config struct {
-	Dir          string        `yaml:"dir"`
-	SyncDir      string        `yaml:"sync_dir"`
-	BlockRanges  time.Duration `yaml:"block_ranges_period"`
-	Retention    time.Duration `yaml:"retention_period"`
-	ShipInterval time.Duration `yaml:"ship_interval"`
-	Backend      string        `yaml:"backend"`
+	Dir            string        `yaml:"dir"`
+	SyncDir        string        `yaml:"sync_dir"`
+	BlockRanges    time.Duration `yaml:"block_ranges_period"`
+	Retention      time.Duration `yaml:"retention_period"`
+	ShipInterval   time.Duration `yaml:"ship_interval"`
+	Backend        string        `yaml:"backend"`
+	ConcurrentSync int           `yaml:"concurrent_sync"`
```

Comment:
I would rename this to `ConcurrentSyncLimit` / `concurrent_sync_limit` / `experimental.tsdb.concurrent-sync-limit`.